compiler/noirc_evaluator/src/ssa/opt/flatten_cfg.rs

```diff
@@ -661,13 +660,11 @@ impl<'f> Context<'f> {
                 // block arguments, it is safe to use the jmpif block here.
                 last_block: jmpif_block,
                 store_values: HashMap::default(),
-                local_allocations: HashSet::new(),
             }
         } else {
             self.push_condition(jmpif_block, new_condition);
             self.insert_current_side_effects_enabled();
             let old_stores = std::mem::take(&mut self.store_values);
-            let old_allocations = std::mem::take(&mut self.local_allocations);
```

Comment:
`local_allocations` inside of `Branch` is unused but this operation does clear `self.local_allocations` which is still being used in the flattening pass. This looks to not be causing any issues though so it may be fine, although  `local_allocations` no longer becomes the allocations local to the current branch?

Reply:
Right, I think we can revert this particular change to be safe.

Reply:
Ah, thanks. I've reverted this change to take the local allocations and replace them after inlining the block.